xgi/generators/nonuniform.py

```diff
@@ -443,3 +445,19 @@ def random_flag_complex(N, p, max_order=2, seed=None):
     S.add_simplices_from(max_cliques, max_order=max_order)
 
     return S
+
+
+@np_random_state("seed")
+def watts_strogatz_hypergraph(n, d, k, l, p, seed=None):
+    H = lattice(n, d, k, l)
```

Comment:
Is there a reference for this and the lattice hypergraph? Both look great!

Reply:
Nope, I was just trying to quickly add something to try out something on WS-like hypergraphs... I kinda want to keep them for now, though undocumented because I haven't really tested these at all.